packages/legend-query/src/components/QueryBuilderPropertySearchPanel.tsx

```diff
@@ -0,0 +1,595 @@
+              : QUERY_BUILDER_EXPLORER_TREE_DND_TYPE.PRIMITIVE_PROPERTY
+            : QUERY_BUILDER_EXPLORER_TREE_DND_TYPE.ROOT,
+        item: (): { node?: QueryBuilderExplorerTreePropertyNodeData } =>
+          node instanceof QueryBuilderExplorerTreePropertyNodeData
+            ? { node }
+            : {},
+        collect: (monitor): { isDragging: boolean } => ({
+          isDragging: monitor.isDragging(),
+        }),
+        end: (): void => {
+          propertySearchPanelState.setIsSearchPanelOpen(true);
+        },
+      }),
+      [node],
+    );
+    const isMultiple =
+      (node instanceof QueryBuilderExplorerTreePropertyNodeData &&
+        (node.property.multiplicity.upperBound === undefined ||
+          node.property.multiplicity.upperBound > 1)) ||
+      (node instanceof QueryBuilderExplorerTreeSubTypeNodeData &&
+        (node.multiplicity.upperBound === undefined ||
+          node.multiplicity.upperBound > 1));
+    queryBuilderState.explorerState.propertySearchPanelState.setIsSearchPanelOpen(
+      !isDragging,
+    );
+    const [isExpandable, setIsExpandable] = useState(false);
+    const nodeExpandIcon = isExpandable ? (
+      <ChevronDownIcon />
+    ) : (
+      <ChevronRightIcon />
+    );
+    const getChildrenNodes = (): QueryBuilderExplorerTreeNodeData[] => {
+      const childNodes: QueryBuilderExplorerTreeNodeData[] = [];
+      if (node.childrenIds.length) {
+        if (
+          (node instanceof QueryBuilderExplorerTreePropertyNodeData ||
+            node instanceof QueryBuilderExplorerTreeSubTypeNodeData) &&
+          node.type instanceof Class
+        ) {
+          (node instanceof QueryBuilderExplorerTreeSubTypeNodeData
+            ? getAllOwnClassProperties(node.type)
+            : getAllClassProperties(node.type)
+          ).forEach((property) => {
+            const propertyTreeNodeData = getQueryBuilderPropertyNodeData(
+              property,
+              node,
+              guaranteeNonNullable(
+                queryBuilderState.querySetupState
+                  .mappingModelCoverageAnalysisResult,
+              ),
+            );
+            if (
+              !(propertyTreeNodeData.type instanceof Class) &&
+              propertyTreeNodeData.mappingData.mapped
+            ) {
+              childNodes.push(propertyTreeNodeData);
+            }
+          });
+        }
+      }
+      return childNodes;
+    };
+    const parentNode = propertySearchPanelState.allMappedPropertyNodes.find(
+      (pn) =>
+        node instanceof QueryBuilderExplorerTreePropertyNodeData &&
+        node.parentId === pn.id,
+    );
+
+    return (
+      <div>
+        <div
+          className="tree-view__node__container"
+          ref={dragConnector}
+          style={{
+            paddingLeft: `${(level - 1) * stepPaddingInRem + 0.5}rem`,
+            display: 'flex',
+          }}
+          onClick={(): void => setIsExpandable(!isExpandable)}
```

Comment:
I believe we can use `onDragStart` here to trigger hidding the panel while dragging.